tests/30rooms/05aliases.pl

```diff
@@ -101,31 +102,119 @@
       });
    };
 
-test "Creators can delete alias",
-   requires => [ $creator_fixture, $room_fixture, room_alias_fixture(),
-                 qw( can_create_room_alias )],
+test "Regular users can add and delete aliases in the default room configuration",
+   requires => [
+      $creator_fixture, $second_user_fixture,
+      magic_room_fixture(
+         requires_users => [ $creator_fixture, $second_user_fixture, ]
+      ),
+      room_alias_fixture(),
+      qw( can_create_room_alias ),
+   ],
 
    proves => [qw( can_delete_room_alias )],
 
    do => sub {
-      my ( $user, $room_id, $room_alias ) = @_;
-      my $server_name = $user->http->server_name;
+      my ( $creator, $other_user, $room_id, $room_alias ) = @_;
+      _test_can_create_and_delete_alias( $room_id, $other_user, $room_alias );
+   };
 
-      do_request_json_for( $user,
-         method => "PUT",
-         uri    => "/r0/directory/room/$room_alias",
 
-         content => { room_id => $room_id },
+test "Regular users can add and delete aliases when m.room.aliases is restricted",
```

Comment:
Will this only pass on old room versions?

Reply:
it should pass on all room versions. The idea of MSC2260 is that you can still add aliases, but if you don't have perms in the room there will be no m.room.aliases event.